### PR TITLE
Replace appmanifest's launch algorithm with one that has launch_handler behaviors

### PR DESCRIPTION
**Pull request by alancutter**
https://github.com/WICG/web-app-launch/issues/67 https://github.com/WICG/web-app-launch/issues/69 https://github.com/WICG/web-app-launch/issues/84

This builds on top of https://github.com/w3c/manifest/pull/1056 which adds a `launch web application` algorithm to the manifest spec. This updates the launch handler spec to replace that algorithm with its own. This enables other webapp specs (not in WICG) to gain the launch_handler behaviour without directly referencing this WICG spec.

Additionally this replaces the concept of "web app launch client" with the existing "application context" concept from the manifest spec which is effectively the same thing.

### CODE REVIEW

**Review by ewilligers — APPROVED** [review]

**Comment by alancutter** [comment]
I'll wait for https://github.com/w3c/manifest/pull/1056 to land before merging this one.

File: index.html
Comment (fallaciousreasoning) [inline index.html:299]: This almost sounds as if the algorithm will only have one of a target url or LaunchParams

**Review by fallaciousreasoning — APPROVED** [review]
LGTM, with one comment

File: index.html
Comment (alancutter) [reply index.html:299]: That is the intention though it might be a bit dodgy to do.

Comment (alancutter) [reply index.html]: Reworded.

**Review by mgiuca — COMMENTED** [review]
Overall looks good, no comments on the code (though see separate bugs [here](https://github.com/WICG/web-app-launch/issues/69#issuecomment-1305015823) and [here](https://github.com/WICG/web-app-launch/issues/84)).

Just some comments on the algorithm structure / redundancy.

File: index.html
Comment (alancutter) [reply index.html]: Deduped.

File: index.html
Comment (alancutter) [reply index.html]: Updated as discussed offline.

File: index.html
Comment (alancutter) [reply index.html]: Removed "web app launch client" in favour of "application context".

File: index.html
Comment (alancutter) [reply index.html]: Removed section.

**Comment by alancutter** [comment]
The manifest exports need a bit of time to propagate through, will hold off landing until then to make sure the xrefs work correctly.

**Comment by alancutter** [comment]
The red squiggles remaining will be fixed after https://github.com/w3c/manifest/pull/1062 lands and the xrefs have time to propagate through.

**Pull request merged**
